utilities/rpp-unittests/OCL/Single_ocl.cpp

```diff
@@ -0,0 +1,533 @@
+    strcat(dst, funcName);
+    mkdir(dst, 0700);
+    strcat(dst, "/");
+
+    DIR *dr = opendir(src);
+    while ((de = readdir(dr)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        noOfImages += 1;
+        break;
+    }
+    closedir(dr);
+
+    RppiSize *srcSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    RppiSize *dstSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+    //const int images = noOfImages;
+    char imageNames[images][1000];
+
+    DIR *dr1 = opendir(src);
+    while ((de = readdir(dr1)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        strcpy(imageNames[count], de->d_name);
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, imageNames[count]);
+        if (ip_channel == 3)
+        {
+            image = imread(temp, 1);
+        }
+        else
+        {
+            image = imread(temp, 0);
+        }
+        srcSize[count].height = image.rows;
+        srcSize[count].width = image.cols;
+        ioBufferSize += (unsigned long long)srcSize[count].height * (unsigned long long)srcSize[count].width * (unsigned long long)ip_channel;
+
+        count++;
+        break;
+    }
+    closedir(dr1);
+
+    Rpp8u *input = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *input_second = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *output = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+
+    /* Read the input image */
+    DIR *dr2 = opendir(src);
+    DIR *dr2_second = opendir(src_second);
+    count = 0;
+    i = 0;
+    while ((de = readdir(dr2)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, de->d_name);
+        char temp_second[1000];
+        strcpy(temp_second, src1_second);
+        strcat(temp_second, de->d_name);
+        if (ip_channel == 3)
+        {
+            image = imread(temp, 1);
+            image_second = imread(temp_second, 1);
+        }
+        else
+        {
+            image = imread(temp, 0);
+            image_second = imread(temp_second, 0);
+        }
+        Rpp8u *ip_image = image.data;
+        Rpp8u *ip_image_second = image_second.data;
+        for (j = 0; j < srcSize[i].height * srcSize[i].width * ip_channel; j++)
+        {
+            input[count] = ip_image[j];
+            input_second[count] = ip_image_second[j];
+            count++;
+        }
+        i++;
+        break;
+    }
+
+    closedir(dr2);
+    cl_mem d_input, d_input_second, d_output;
+    cl_platform_id platform_id;
+    cl_device_id device_id;
+    cl_context theContext;
+    cl_command_queue theQueue;
+    cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)